src/app/shared/components/addregressionregion/addregressionregion.component.ts

```diff
@@ -0,0 +1,437 @@
+      center: new L.LatLng(39.8283, -98.5795),
+      zoom: 4,
+      layers: [tileLayer_streets]
+    });
+
+    // Create basemap collection
+    const baseMaps = {
+      'Streets': tileLayer_streets,
+      'Topography': tileLayer_topography,
+      'Grayscale': tileLayer_grayscale,
+      'Satellite': tileLayer_satellite
+    };
+
+    // Add basemap control to the map
+    L.control.layers(baseMaps).addTo(this.map);
+
+  }
+
+  public getRegRegions() {
+    // re-query regression regions after new one is added
+    this._settingsService.getEntities(this.configSettings.regionURL + this.selectedRegion.id + '/' + this.configSettings.regRegionURL)
+      .subscribe(res => {
+        if (res.length > 1) { res.sort((a, b) => a.name.localeCompare(b.name)); }
+        this.regressionRegions = res;
+      });
+  }
+
+  public showModal(): void {
+    if (this.selectedRegion) { this.getRegRegions(); }
+    this.modalRef = this._modalService.open(this.modalElement, { backdrop: 'static', keyboard: false, size: 'lg' });
+    this.modalRef.result.then(
+      result => {
+        // this is the solution for the first modal losing scrollability
+        if (document.querySelector('body > .modal')) {
+          document.body.classList.add('modal-open');
+        }
+        this.CloseResult = `Closed with: ${result}`;
+      },
+      reason => {
+        this.CloseResult = `Dismissed ${this.getDismissReason(reason)}`;
+        this.cancelCreateRegression();
+      }
+    );
+  }
+
+  outputWimMessages(msg) {
+    // output messages from http request to toast
+    for (const key of Object.keys(msg)) {
+      for (const item of msg[key]) {
+        this._toasterService.pop(key, key.charAt(0).toUpperCase() + key.slice(1), item);
+      }
+    }
+  }
+
+  private getDismissReason(reason: any): string {
+    if (reason === ModalDismissReasons.ESC) return 'by pressing ESC';
+    else if (reason === ModalDismissReasons.BACKDROP_CLICK) return 'by clicking on a backdrop';
+    else return `with: ${reason}`;
+  }
+
+  ngOnDestroy() {
+    this.modalSubscript.unsubscribe();
+  }
+
+  // show add regression region modal
+  public showNewRegressionRegionForm(rr?) {
+    // shows form for creating new regression or editing regression 
+    if (rr) { // edit existing regression region
+      this._loaderService.showFullPageLoad();
+      this.addRegReg = false;
+      this._settingsService.getEntities(this.configSettings.regRegionURL + '/' + rr + '?includeGeometry=true')
+        .subscribe((res) => {
+          this.selectedRegRegion = res;
+          // set values in add regression region modal
+          this.newRegRegForm.controls['name'].setValue(this.selectedRegRegion.name);
+          this.newRegRegForm.controls['description'].setValue(this.selectedRegRegion.description);
+          this.newRegRegForm.controls['code'].setValue(this.selectedRegRegion.code);
+          this.newRegRegForm.controls['location'].setValue(this.selectedRegRegion.location);
```

Comment:
I don't love all this, but I think I'll make an enhancement issue for cleaning up more.